### PR TITLE
feat(petrinet): add script for AlgebraicPetri conversion

### PR DESCRIPTION
**Pull request by mehalter**

### CODE REVIEW

File: petrinet/scripts/julia/test.jl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+include("./ASKEMPetriNets.jl")
+using AlgebraicPetri
+using Catlab.CategoricalAlgebra, Catlab.Graphics
+using JSON
+
+# Load a model
+askemnet = ASKEMPetriNets.to_petri("../../examples/sir.json")
+
+askemnet.model |> to_graphviz
```
Comment (liunelson) [inline petrinet/scripts/julia/test.jl:12]: I get this error from Line 12. Not a big deal, probably versioning of `Catlab.Graphics`
Screenshot 2023-06-20 at 10 17 26

File: petrinet/scripts/julia/test.jl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ASKEMPetriNets.update!(askemnet)
+
+# Print new JSON
+JSON.print(askemnet, 2)
+
+typed_askemnet = ASKEMPetriNets.to_typed_petri("../../examples/sir_typed.json")
+
+typed_askemnet.model |> to_graphviz
+
+@assert is_natural(typed_askemnet.model)
```
Comment (liunelson) [inline petrinet/scripts/julia/test.jl:28]: I get this error from Line 28. Not sure what this assertion is testing.

Screenshot 2023-06-20 at 10 19 34

**Comment by liunelson** [comment]
Does `ASKEMPetriNet` and `TypedASKEMPetriNet` (which I understand to be the Catlab's representation of AMR models) support this conversion to SciML/EMA's `ODESystem`?
https://github.com/DARPA-ASKEM/experiments/blob/main/notebooks/Nelson/model-repr-conversion/petri-to-equations.ipynb

I get this error:
Screenshot 2023-06-20 at 10 32 16

I imagine that there is a more current way than to use a Labelled Bilayer as an intermediate step?

**Comment by mehalter** [comment]
@liunelson I have updated the commit with compat entries, it looks like you probably don't have the correct versions of things. Now the project will handle telling you if there are version issues in your environment

File: petrinet/scripts/julia/test.jl (hunk 1 above)
Comment (mehalter) [reply petrinet/scripts/julia/test.jl:12]: Yeah, I updated the Project.toml with compat entries to help you get the correct versions of things

File: petrinet/scripts/julia/test.jl (hunk 2 above)
Comment (mehalter) [reply petrinet/scripts/julia/test.jl:28]: similarly probably coming from version mismatches

**Comment by mehalter** [comment]
@liunelson Yeah you definitely can, the main thing is you need to do the migration on the catlab structures in the ASKEMNet.  So if you have an `ASKEMPetriNet` then you want to call `migrate!(model_bilayer, askemnet.model)` and if you have a `TypedASKEMPetriNet` then you want to call `migrate!(model_bilayer, typed_askemnet.model.dom)` to get the domain of the `ACSetTransformation` that is the model.

**Comment by liunelson** [comment]
@mehalter 
Thanks for the explanation. I just ran `] up` but I get the same errors, probably because my Julia is on `0.8` and `AlgebraicPetri` and `Catlab` couldn't update (according to `] status --outdated`.

Was there a particular reason for switching to Julia 0.9?

New EMA error now 😅 
Screenshot 2023-06-20 at 12 00 42

**Comment by mehalter** [comment]
@liunelson We had to move to Julia v1.9 because there were several stack overflow bugs in Julia <v1.9 that were blocking our development making Catlab completely unusable basically.

Also yeah looks like EasyModelAnalysis has some issues that need to be resolved. Seems like a decent fix, looks like they are conflicting on some of SciML's exports from a couple packages that export things of the same name.

**Comment by mehalter** [comment]
@liunelson this is probably coming from a recent update, you could probably investigate and find an older version to add rather than the latest. The latest may just be broken.

**Review by liunelson — APPROVED** [review]

**Pull request merged**